### PR TITLE
Improve plot selection support

### PR DESCRIPTION
**Pull request by ritch**
## Summary by CodeRabbit

- **New Features**
  - Enhanced event handling in the `PlotlyView` component, allowing for more detailed data retrieval during user interactions.
  - Introduced new event handler parameters (`on_click`, `on_selected`, `on_double_click`) for greater customization in visualizations.

- **Documentation**
  - Expanded documentation for event handler parameters, including usage examples to improve clarity and usability for developers. 

- **Bug Fixes**
  - Improved handling of specific event cases, ensuring that empty strings are treated appropriately during event processing.

### CODE REVIEW

File: app/packages/app/src/useEvents/useEvents.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ const useEvents = (
   return {
     subscriptions,
     handler: useCallback((event: string, payload: string) => {
-      if (event === "ping") {
+      if (event === "ping" || event === "") {
```
Comment (ritch) [inline app/packages/app/src/useEvents/useEvents.ts:30]: @benjaminpkane I can't get the app to run without this

File: app/packages/core/src/components/Grid/Grid.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ import { v4 as uuid } from "uuid";
 import { gridCrop, gridSpacing, pageParameters } from "./recoil";
 import useAt from "./useAt";
 import useEscape from "./useEscape";
+import useRecords from "./useRecords";
```
Comment (ritch) [inline app/packages/core/src/components/Grid/Grid.tsx:20]: TODO: Remove

File: fiftyone/operators/types.py [hunk 3]
```diff
@@ -1483,10 +1483,51 @@ class PlotlyView(View):
...
+        def render(self, ctx):
+            panel.plot("my_plot", on_click=self.on_click, on_selected=self.on_selected)
+
+        def print_params(self, ctx, params):
+            for key, value in params.items():
+                ctx.print(f"{key}: {value}")
+
+        def on_click(self, ctx):
+            # available params
+            self.print_prams(ctx, {
```
Comment (ritch) [inline fiftyone/operators/types.py:1497]: Is there a better way to document these?

File: app/packages/core/src/plugins/SchemaIO/components/PlotlyView.tsx [hunk 4]
```diff
@@ -9,6 +9,30 @@ import { HeaderView } from ".";
...
 
+type TraceWithIds = {
+  name?: string;
+  ids?: string[];
+};
+
+function getIdForTrace(
+  point: Plotly.Point,
+  trace: TraceWithIds,
+  options: { is2DArray?: boolean } = {}
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/PlotlyView.tsx:20]: Might be simpler to just use duck typing to figure out if its 2d or not...

**Comment by ritch** [comment]
@imanjra - if you have suggestions on how to add tests for these components LMK - I feel like this should come with a few tests but its not clear where/how at this point.

**Comment by imanjra** [comment]
PlotlyView indeed has lot of util logic. Perhaps, it may be worth moving em to `.../PlotlyView/utils.ts` and unit testing em? 

For integrated tests, E2E may be the easier route:
- Plugin: https://github.com/voxel51/fiftyone/tree/develop/e2e-pw/src/shared/assets/plugins/e2e
- Example Python Panel test: https://github.com/voxel51/fiftyone/blob/develop/e2e-pw/src/oss/specs/operators/python-panels-grid.spec.ts#L42

File: app/packages/core/src/plugins/SchemaIO/components/PlotlyView.tsx [hunk 5]
```diff
@@ -61,30 +89,44 @@ export default function PlotlyView(props: ViewPropsType) {
...
     const defaultParams = {
+      id,
       path: props.path,
       relative_path: props.relativePath,
       schema: props.schema,
       view,
       event,
       value,
       label,
+      shift_pressed: Boolean(e?.event?.shiftKey),
```
Comment (imanjra) [inline app/packages/core/src/plugins/SchemaIO/components/PlotlyView.tsx:100]: rename `shift_pressed` to `multiselect` (or `multiple`) if it's synonymous?

**Review by imanjra — APPROVED** [review]
LGTM

**Pull request merged**